### PR TITLE
Add OPTIONS to list of default allowed methods

### PR DESCRIPTION
**Pull request by rcchen**
This covers situations where an OPTIONS preflight request is made.

### CODE REVIEW

File: src/main/java/com/palantir/dropwizard/websecurity/WebSecurityBundle.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
     /**
      * The default value of CORS Allowed Methods. It includes commonly used methods.
      */
-    public static final String DEFAULT_ALLOWED_METHODS = "DELETE,GET,HEAD,POST,PUT";
+    public static final String DEFAULT_ALLOWED_METHODS = "DELETE,GET,HEAD,POST,PUT,OPTIONS";
```
Comment (derekcicerone-zz) [inline src/main/java/com/palantir/dropwizard/websecurity/WebSecurityBundle.java]: alpha, also i think this needs to be updated in README.MD

**Comment by derekcicerone-zz** [comment]
This looks good to me. @rcchen just to clarify, you tried this without the change and it doesn't work, right?

**Comment by rcchen** [comment]
It was an issue that we ran into a couple months ago during Permissions.Dialog development so I included it preemptively; I can test it now.

**Comment by derekcicerone-zz** [comment]
Please check because we are using a standard CORS filter and since OPTIONS requests are part of CORS it seems reasonable that the defaults would work. Also, Ben tested with this and it worked for him so its a little weird that it didn’t work for you.

**Comment by rcchen** [comment]
I need to run so I'll have to confirm later, but I think I accidentally conflated Gatekeeper and Multipass requirements. Specifically, persisting policies to Gatekeeper is done through a PUT request that triggers an OPTIONS preflight

> In particular, a request is preflighted if: It uses methods other than GET, HEAD or POST.  Also, if POST is used to send request data with a Content-Type other than application/x-www-form-urlencoded, multipart/form-data, or text/plain, e.g. if the POST request sends an XML payload to the server using application/xml or text/xml, then the request is preflighted.
- https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#Preflighted_requests

So this is NOT necessary for Multipass but IS necessary for Gatekeeper, which should be on this bundle eventually.

**Comment by derekcicerone-zz** [comment]
CORS requests need to support OPTIONS pre-flight requests, that is for sure. The only question I have is whether we need to explicitly list it as an allowed method or if the filter just always allows OPTIONS requests (its hardcoded). Can you try a PUT request without modifying the config? If we need to add it to the defaults thats totally cool, just wanna make sure we actually try out a pre-flight scenario to verify this change is necessary.

**Comment by jmcampanini** [comment]
// for SA: @tomshen 

here's the primary handler for [`CrossOriginFilter`](https://github.com/eclipse/jetty.project/blob/master/jetty-servlets/src/main/java/org/eclipse/jetty/servlets/CrossOriginFilter.java#L260-L265)

my understanding of the basic logic is:
1. if it's a "simple" method (hard-coded to `GET`, `POST`, `HEAD`), call [`handleSimpleResponse`](https://github.com/eclipse/jetty.project/blob/master/jetty-servlets/src/main/java/org/eclipse/jetty/servlets/CrossOriginFilter.java#L396-L406)
2. else if it's a preflight request (method is `OPTIONS` and has a specific header field req'd for preflight), call `handlePreflight`
3. else, call [`handleSimpleResponse`](https://github.com/eclipse/jetty.project/blob/master/jetty-servlets/src/main/java/org/eclipse/jetty/servlets/CrossOriginFilter.java#L396-L406) anyway

my read is the `allowedMethods` values do not impact whether or not a request is a preflight request, so there is no need to add `OPTIONS` for preflight requests. specifically, it's not used in the [`isPreflightRequest`](https://github.com/eclipse/jetty.project/blob/master/jetty-servlets/src/main/java/org/eclipse/jetty/servlets/CrossOriginFilter.java#L386-L394) method used in step 2 above

given that, not sure if this PR is necessary, but waiting for @rcchen to verify

**Comment by rcchen** [comment]
Just checked, and leaving off `OPTIONS` has no impact on whether a PUT request successfully preflights or not. Closing.

**Pull request closed without merging**